From 98c5fa349ca64fb6e6a4c29a32f8e3ced6fb81b5 Mon Sep 17 00:00:00 2001
From: Etienne Cordonnier <ecordonnier@snap.com>
Date: Fri, 7 Jun 2024 00:06:42 +0200
Subject: [PATCH] fix wrong libtool path

Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Dominik Schnitzer <dominik@snap.com>
---
 configure | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/configure b/configure
index a7c6d72..c2c1d6a 100755
--- a/configure
+++ b/configure
@@ -264,15 +264,7 @@ if test "$gcc" -eq 1 && ($cc -c $test.c) >> configure.log 2>&1; then
         SHAREDLIB=libz$shared_ext
         SHAREDLIBV=libz.$VER$shared_ext
         SHAREDLIBM=libz.$VER1$shared_ext
-        LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"}
-        if "${CROSS_PREFIX}libtool" -V 2>&1 | grep Apple > /dev/null; then
-            AR="${CROSS_PREFIX}libtool"
-        elif libtool -V 2>&1 | grep Apple > /dev/null; then
-            AR="libtool"
-        else
-            AR="/usr/bin/libtool"
-        fi
-        ARFLAGS="-o" ;;
+        LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"} ;;
   *)
         LDSHARED=${LDSHARED-"$cc -shared"} ;;
   esac
-- 
2.36.1.vfs.0.0

