From 3b7dfa06c12eb605ae244bbdf37a5c502f63a035 Mon Sep 17 00:00:00 2001
From: Dominik Schnitzer <dominik@snap.com> 
Date: Fri, 12 May 2023 10:11:56 +0000
Subject: [PATCH] fix wrong libtool path

Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Dominik Schnitzer <dominik@snap.com>
---
 configure | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/configure b/configure
index 69dfa3f..dcd0e75 100755
--- a/configure
+++ b/configure
@@ -240,13 +240,7 @@ if test "$gcc" -eq 1 && ($cc -c $test.c) >> configure.log 2>&1; then
              SHAREDLIB=libz$shared_ext
              SHAREDLIBV=libz.$VER$shared_ext
              SHAREDLIBM=libz.$VER1$shared_ext
-             LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"}
-             if libtool -V 2>&1 | grep Apple > /dev/null; then
-                 AR="libtool"
-             else
-                 AR="/usr/bin/libtool"
-             fi
-             ARFLAGS="-o" ;;
+             LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"} ;;
   *)             LDSHARED=${LDSHARED-"$cc -shared"} ;;
   esac
 else
-- 
2.36.1.vfs.0.0

