Index: apple-x86-odcctools-758.159/ld64/src/Options.cpp
===================================================================
--- apple-x86-odcctools-758.159.orig/ld64/src/Options.cpp	2013-09-03 21:54:26.325145025 +0000
+++ apple-x86-odcctools-758.159/ld64/src/Options.cpp	2013-09-03 21:54:26.581145019 +0000
@@ -30,6 +30,10 @@
 #include <ctype.h>
 #include <vector>
 #include <cstdio>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <limits.h>
 
 #include "config.h"
 #include "Options.h"
Index: apple-x86-odcctools-758.159/ld64/Makefile.in
===================================================================
--- apple-x86-odcctools-758.159.orig/ld64/Makefile.in	2013-09-03 21:54:26.325145025 +0000
+++ apple-x86-odcctools-758.159/ld64/Makefile.in	2013-09-03 21:54:26.581145019 +0000
@@ -47,7 +47,7 @@
 		  -DASPROG="\"$(ASPROG)\"" $(LTO_DEFS)
 
 MYLDFLAGS	= -L$(top_builddir)/libstuff
-MYLIBS		= -lstuff $(SSL_LIBS) $(UUID_LIBS) $(LTO_LIBS)
+MYLIBS		= -lstuff $(SSL_LIBS) $(UUID_LIBS) $(LTO_LIBS) -ldl
 
 MYCOMPILEFLAGS	= $(WARNINGS) $(MYWARNINGS) $(DEFS) $(MYDEFS) \
 	$(CPPFLAGS) $(MYINCLUDES) $(CFLAGS) $(MDYNAMICNOPIC) 
Index: apple-x86-odcctools-758.159/configure.ac
===================================================================
--- apple-x86-odcctools-758.159.orig/configure.ac	2011-03-31 17:08:42.000000000 +0000
+++ apple-x86-odcctools-758.159/configure.ac	2013-09-03 22:28:45.781095869 +0000
@@ -284,7 +284,7 @@
 )
 
 AC_CHECK_FUNCS([strmode qsort_r qsort getattrlist NSIsSymbolNameDefined strlcpy strlcat])
-AC_CHECK_HEADERS([CommonCrypto/CommonDigest.h],[],[SSL_LIBS=-lssl])
+AC_CHECK_HEADERS([CommonCrypto/CommonDigest.h],[],[SSL_LIBS=-lcrypto])
 AC_SUBST(SSL_LIBS)
 
 AC_CHECK_DECLS([getc_unlocked, strlcpy, strlcat])
