Index: ar/ar.c
===================================================================
--- ar/ar.c	(revision 287)
+++ ar/ar.c	(working copy)
@@ -82,6 +82,7 @@
 
 #include "archive.h"
 #include "extern.h"
+#include "stuff/allocate.h"
 #include "stuff/execute.h"
 #include "stuff/unix_standard_mode.h"
 
Index: ar/contents.c
===================================================================
--- ar/contents.c	(revision 287)
+++ ar/contents.c	(working copy)
@@ -75,6 +75,7 @@
 #include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
+#include <time.h>
 #include <unistd.h>
 
 #include "archive.h"
