--- apple-x86-odcctools-758.159/include/foreign/mach/machine/vm_types.h		2011-04-07 11:27:03.605622001 +0200
+++ apple-x86-odcctools-758.159-patched/include/foreign/mach/machine/vm_types.h	2011-04-07 11:27:12.951666000 +0200
@@ -6,8 +6,13 @@
 typedef uint32_t        natural_t;
 typedef int32_t         integer_t;
 
+#ifdef __i386__
 typedef natural_t       vm_offset_t;
 typedef natural_t       vm_size_t;
+#else
+typedef uint64_t        vm_offset_t;
+typedef uint64_t        vm_size_t;
+#endif
 
 typedef uint64_t        mach_vm_address_t;
 typedef uint64_t        mach_vm_offset_t;
