diff -ru gcc-9.3.0-orig/gcc/config/darwin-c.c gcc-9.3.0/gcc/config/darwin-c.c
--- gcc-9.3.0-orig/gcc/config/darwin-c.c	2020-11-01 10:34:44.147351721 +0100
+++ gcc-9.3.0/gcc/config/darwin-c.c	2020-11-01 10:35:55.582205583 +0100
@@ -480,8 +480,8 @@
      /* See if our directory starts with the standard prefix.
 	 "Translate" them, i.e. replace /usr/local/lib/gcc... with
 	 IPREFIX and search them first.  */
-      if (iprefix && (len = cpp_GCC_INCLUDE_DIR_len) != 0
-	  && !strncmp (fname, cpp_GCC_INCLUDE_DIR, len))
+      if (iprefix && (len = strlen(GCC_INCLUDE_DIRVAR) - 7) != 0
+	  && !strncmp (fname, GCC_INCLUDE_DIRVAR, len))
 	{
 	  str = concat (iprefix, fname + len, NULL);
 	  add_path (str, INC_SYSTEM, /*c++aware=*/true, false);
diff -ru gcc-9.3.0-orig/gcc/config/darwin.h gcc-9.3.0/gcc/config/darwin.h
--- gcc-9.3.0-orig/gcc/config/darwin.h	2020-11-01 10:34:44.147351721 +0100
+++ gcc-9.3.0/gcc/config/darwin.h	2020-11-01 10:35:55.582205583 +0100
@@ -259,10 +259,6 @@
    isysroot is specified.  */
 #define LINK_SYSROOT_SPEC "%{isysroot*:-syslibroot %*}"
 
-/* Suppress the addition of extra prefix paths when a sysroot is in use.  */
-#define STANDARD_STARTFILE_PREFIX_1 ""
-#define STANDARD_STARTFILE_PREFIX_2 ""
-
 /* Please keep the random linker options in alphabetical order (modulo
    'Z' and 'no' prefixes). Note that options taking arguments may appear
    multiple times on a command line with different arguments each time,
