From 8f7f5820b24f57460e26440e770c8407ec11e10d Mon Sep 17 00:00:00 2001
From: Etienne Cordonnier <ecordonnier@snap.com>
Date: Thu, 11 Aug 2022 15:39:44 +0200
Subject: [PATCH] change macosx-version-min to 12.3

Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Etienne Cordonnier <ecordonnier@snap.com>
---
 libgcc/config/t-darwin | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libgcc/config/t-darwin b/libgcc/config/t-darwin
index a3bb70c6a0a..8923a375c68 100644
--- a/libgcc/config/t-darwin
+++ b/libgcc/config/t-darwin
@@ -1,5 +1,8 @@
 # Set this as a minimum (unless overriden by arch t-files) since it's a
 # reasonable lowest common denominator that works for all our archs.
+DARWIN_MIN_CRT_VERSION="-mmacosx-version-min=12.3"
+DARWIN_MIN_LIB_VERSION="-mmacosx-version-min=12.3"
+
 HOST_LIBGCC2_CFLAGS += $(DARWIN_MIN_LIB_VERSION)
 
 crt3.o: $(srcdir)/config/darwin-crt3.c
-- 
2.36.1.vfs.0.0

