diff --git a/clang/lib/Driver/ToolChains/Darwin.cpp b/clang/lib/Driver/ToolChains/Darwin.cpp
index 46265c1b9f1..008d86ac610 100644
--- a/clang/lib/Driver/ToolChains/Darwin.cpp
+++ b/clang/lib/Driver/ToolChains/Darwin.cpp
@@ -1998,12 +1998,18 @@ void DarwinClang::AddClangCXXStdlibIncludeArgs(
     case llvm::Triple::x86:
     case llvm::Triple::x86_64:
       IsBaseFound = AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
-                                                "4.2.1",
-                                                "i686-apple-darwin10",
-                                                arch == llvm::Triple::x86_64 ? "x86_64" : "");
+                                                "9.3.0",
+                                                "x86_64-pokysdk-darwin9",
+                                                "");
       IsBaseFound |= AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
-                                                "4.0.0", "i686-apple-darwin8",
-                                                 "");
+                                                "9.3.0", "", "");
+      {
+	llvm::SmallString<128> Runtime = Sysroot;
+	llvm::sys::path::append(Runtime, "usr", "local", "pokysdk-x86_64");
+	llvm::sys::path::append(Runtime, "sysroots", "x86_64-pokysdk-darwin9", "runtime");
+	llvm::sys::path::append(Runtime, "usr", "include");
+      	addSystemInclude(DriverArgs, CC1Args, Runtime);
+      }
       break;
 
     case llvm::Triple::arm:
