From cb9b87097803b4444ec9aa52b5054ba88e4da01e Mon Sep 17 00:00:00 2001
From: Dominik Schnitzer <dominik@snap.com>
Date: Wed, 27 Apr 2022 11:22:17 +0000
Subject: [PATCH] Fixes for Darwin SDKs

Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Dominik Schnitzer <dominik@snap.com>
Index: llvm-project-18.1.5.src/clang/lib/Driver/ToolChains/Darwin.cpp
===================================================================
--- llvm-project-18.1.5.src.orig/clang/lib/Driver/ToolChains/Darwin.cpp
+++ llvm-project-18.1.5.src/clang/lib/Driver/ToolChains/Darwin.cpp
@@ -2583,12 +2583,19 @@ void DarwinClang::AddClangCXXStdlibInclu
     case llvm::Triple::x86:
     case llvm::Triple::x86_64:
       IsBaseFound = AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
-                                                "4.2.1",
-                                                "i686-apple-darwin10",
-                                                arch == llvm::Triple::x86_64 ? "x86_64" : "");
+                                                "13.4.0",
+                                                "x86_64#SDK_VENDOR#-darwin21",
+                                                "");
       IsBaseFound |= AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
-                                                "4.0.0", "i686-apple-darwin8",
-                                                 "");
+                                                "13.4.0", "", "");
+      {
+          const char *S = ::getenv("YOCTO_SDKPATH");
+          if (S && (S[0] != '\0')) {
+              llvm::SmallString<128> Runtime;
+              llvm::sys::path::append(Runtime, S);
+              addSystemInclude(DriverArgs, CC1Args, Runtime);
+          }
+      }
       break;
 
     case llvm::Triple::arm:
@@ -2605,9 +2612,19 @@
 
     case llvm::Triple::aarch64:
       IsBaseFound = AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
-                                                "4.2.1",
-                                                "arm64-apple-darwin10",
+                                                "13.4.0",
+                                                "aarch64#SDK_VENDOR#-darwin21",
                                                 "");
+      IsBaseFound |= AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
+                                                "13.4.0", "", "");
+      {
+          const char *S = ::getenv("YOCTO_SDKPATH");
+          if (S && (S[0] != '\0')) {
+              llvm::SmallString<128> Runtime;
+              llvm::sys::path::append(Runtime, S);
+              addSystemInclude(DriverArgs, CC1Args, Runtime);
+          }
+      }
       break;
     }
 
Index: llvm-project-18.1.5.src/llvm/cmake/modules/AddLLVM.cmake
===================================================================
--- llvm-project-18.1.5.src.orig/llvm/cmake/modules/AddLLVM.cmake
+++ llvm-project-18.1.5.src/llvm/cmake/modules/AddLLVM.cmake
@@ -2368,7 +2368,6 @@ function(llvm_setup_rpath name)
   endif()
 
   if (APPLE)
-    set(_install_name_dir INSTALL_NAME_DIR "@rpath")
     set(_install_rpath "@loader_path/../lib${LLVM_LIBDIR_SUFFIX}" ${extra_libdir})
   elseif(${CMAKE_SYSTEM_NAME} MATCHES "AIX" AND BUILD_SHARED_LIBS)
     # $ORIGIN is not interpreted at link time by aix ld.
Index: llvm-project-18.1.5.src/llvm/cmake/modules/CrossCompile.cmake
===================================================================
--- llvm-project-18.1.5.src.orig/llvm/cmake/modules/CrossCompile.cmake
+++ llvm-project-18.1.5.src/llvm/cmake/modules/CrossCompile.cmake
@@ -72,7 +72,7 @@ function(llvm_create_cross_target projec
   endif()
 
   add_custom_command(OUTPUT ${${project_name}_${target_name}_BUILD}/CMakeCache.txt
-    COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"
+    COMMAND ${CMAKE_COMMAND} -E env --unset=LDFLAGS --unset=CFLAGS --unset=CXXFLAGS --unset=CC --unset=LD --unset=CXX ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"
         -DCMAKE_MAKE_PROGRAM="${CMAKE_MAKE_PROGRAM}"
         -DCMAKE_C_COMPILER_LAUNCHER="${CMAKE_C_COMPILER_LAUNCHER}"
         -DCMAKE_CXX_COMPILER_LAUNCHER="${CMAKE_CXX_COMPILER_LAUNCHER}"
