diff -ru openssl-1.1.1g-orig/Configurations/10-main.conf openssl-1.1.1g/Configurations/10-main.conf
--- openssl-1.1.1g-orig/Configurations/10-main.conf	2020-11-09 10:11:39.933127687 +0000
+++ openssl-1.1.1g/Configurations/10-main.conf	2020-11-09 08:43:58.943821217 +0000
@@ -1557,6 +1557,30 @@
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         perlasm_scheme   => "macosx",
     },
+    "darwin9-x86_64" => {
+        inherit_from     => [ asm("x86_64_asm") ],
+        CC               => "gcc",
+        CXX              => "g++",
+        lib_cppflags     => add("-DL_ENDIAN"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG",
+        perlasm_scheme   => "macosx",
+        CFLAGS           => picker(default => "-Wall -m64",
+                                   debug   => "-O0 -g",
+                                   release => "-O3"),
+        CXXFLAGS         => picker(default => "-Wall -m64",
+                                   debug   => "-O0 -g",
+                                   release => "-O3"),
+        cflags           => threads("-pthread"),
+        lflags           => "-Wl,-search_paths_first",
+        cxxflags         => combine("-std=c++11", threads("-pthread")),
+        lib_cppflags     => "-DL_ENDIAN",
+        ex_libs          => add("-ldl", threads("-pthread")),
+        thread_scheme    => "pthreads",
+        dso_scheme       => "dlfcn",
+        shared_target    => "darwin-shared",
+        shared_cflag     => "-fPIC",
+        shared_extension => ".\$(SHLIB_VERSION_NUMBER).dylib",
+    },
 
 ##### GNU Hurd
     "hurd-x86" => {
