Index: gtk+-2.12.7/tests/Makefile.am
===================================================================
--- gtk+-2.12.7.orig/tests/Makefile.am	2008-02-11 17:38:25.000000000 +0000
+++ gtk+-2.12.7/tests/Makefile.am	2008-02-11 17:42:59.000000000 +0000
@@ -28,7 +28,6 @@
 
 noinst_PROGRAMS =			\
 	autotestfilechooser		\
-	autotestkeywords		\
 	floatingtest			\
 	buildertest			\
 	simple				\
@@ -91,19 +90,6 @@
 	testtooltips			\
 	testvolumebutton
 
-if HAVE_CXX
-noinst_PROGRAMS += autotestkeywords
-TESTS += autotestkeywords
-
-autotestkeywords_SOURCES = autotestkeywords.cc
-autotestkeywords_CPPFLAGS = -I$(srcdir)/dummy-headers
-
-if HAVE_OBJC
-autotestkeywords_CPPFLAGS += -DHAVE_OBJC=1
-autotestkeywords_CXXFLAGS = -x objective-c++
-endif
-
-endif
 
 autotestfilechooser_DEPENDENCIES = $(TEST_DEPS)
 simple_DEPENDENCIES = $(TEST_DEPS)
